src/server/handlers/consents/{ID}/generateChallenge.ts

```diff
@@ -0,0 +1,130 @@
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files
+ are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+
+ --------------
+ ******/
+
+/* istanbul ignore file */
+
+/*
+ * This flag is to ignore BDD testing
+ * which will be addressed in the future in
+ * ticket #354
+ */
+
+import {
+  updateConsentCredential,
+  isConsentRequestInitiatedByValidSource,
+  putConsentId, ConsentCredential
+} from '../../../../domain/consents/{ID}/generateChallenge'
+import * as challenge from '../../../../lib/challenge'
+import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
+import { consentDB, scopeDB } from '../../../../lib/db'
+import { Consent } from '../../../../model/consent'
+import Logger from '@mojaloop/central-services-logger'
+import { convertScopesToExternal } from '../../../../lib/scopes'
+import { Scope } from '../../../../model/scope'
+
+// Asynchronously deals with generating challenge, updating consent db
+//  and making outgoing PUT consent/{ID} call
+export async function generateChallengeAndPutConsentId (
+  request: Request,
+  consent: Consent,
+  id: string
+): Promise<void> {
+  try {
+    // If there is no pre-existing challenge for the consent id
+    // Generate one and update database
+    if (!consent.credentialChallenge) {
+      // Challenge generation
+      const challengeValue = await challenge.generate()
+
+      // Updating credentials with generated challenge
+      const credential: ConsentCredential = {
+        credentialChallenge: challengeValue,
+        credentialStatus: 'PENDING',
+        credentialType: 'FIDO'
+      }
+
+      consent = await updateConsentCredential(consent, credential)
+    } else if (consent.credentialStatus === 'ACTIVE') {
+      // TODO: Error handling here - dealt with in #355
+      Logger.error('ACTIVE credential consent has requested challenge')
```

Comment:
we should throw here NotImplementedYetError, elsewhere there is a risk we will not implement it

Reply:
same comment as below - re: handling of errors @eoln